subsys/diff/src/main/java/org/commonjava/indy/subsys/diff/RepoChangeHandler.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (C) 2013 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.subsys.diff;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.github.difflib.DiffUtils;
+import com.github.difflib.UnifiedDiffUtils;
+import com.github.difflib.algorithm.DiffException;
+import com.github.difflib.patch.Patch;
+import org.commonjava.indy.change.event.ArtifactStorePreUpdateEvent;
+import org.commonjava.indy.data.IndyDataException;
+import org.commonjava.indy.model.change.RepositoryChangeLog;
+import org.commonjava.indy.model.core.ArtifactStore;
+import org.commonjava.indy.model.core.io.IndyObjectMapper;
+import org.commonjava.indy.subsys.diff.cache.RepoChangelogCache;
+import org.commonjava.indy.subsys.infinispan.CacheHandle;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+
+@ApplicationScoped
+public class RepoChangeHandler
+{
+    private static final int DIFF_PATCH_CONTEXT_LINES = 3;
+
+    @Inject
+    private IndyObjectMapper objectMapper;
+
+    @Inject
+    @RepoChangelogCache
+    private CacheHandle<String, RepositoryChangeLog> repoChangelogCache;
+
+    public void generateRepoChangeLog( @Observes ArtifactStorePreUpdateEvent event )
+    {
+        Collection<ArtifactStore> stores = event.getChanges();
+
+        //TODO: think about how to handle these exceptions
+        List<IndyDataException> exceptions = new ArrayList<>();
```

Comment:
For now I think it's fine to log them to the error log level. I'm not sure what else we can do, or should do.

Reply:
Done with log error way.